engines/mohawk/myst_state.cpp

```diff
@@ -227,6 +229,7 @@ void MystGameState::updateMetadateForSaving(const Common::String &desc) {
 	_metadata.saveMinute = t.tm_min;
 	_metadata.saveDescription = desc;
 	_metadata.totalPlayTime = _vm->getTotalPlayTime();
+	_metadata.autoSave = isAutoSaveAllowed() && (slot == Mohawk::kAutoSaveSlot);
```

Comment:
I don't think it's necessary to call `isAutoSaveAllowed` here. The argument `int slot` could be changed to `bool autosave` and passed by the callers up to the autosave method (Calling  `_gameState->save` instead of `saveGameState` in `autosave`).